CMakeLists.txt

```diff
@@ -1802,6 +1802,7 @@ if(NOT WIN32)
   # ############################################################################
   if(NOT MSVC)
     hpx_add_compile_flag_if_available(-pthread)
+    hpx_add_link_flag_if_available(-pthread TARGETS EXE SHARED)
```

Comment:
This was one of the stranger things I've seen in a while.

There were a few failing tests with the following error message:
```
terminate called after throwing an instance of 'std::system_error'
  what():  Unknown error -1
```
That error was thrown from `std::call_once` because the tests were not (correctly?) linked to `libpthread`. `ldd` did show `libpthread` but in a different place compared to master and the tests themselves were not directly linked to `libpthread`.

Adding a `#include <boost/asio.hpp>` with no other changes to the failing tests make them run again. I've added `-pthread` conditionally as a linker flag here which seems to be the safest option. Linking explicitly to `libpthread` with `-lpthread` would also be an option, but that also has to be conditional.

I sure hope this isn't a case of the Cray clang compiler being "smart". The same does not seem to happen on the vanilla clang configurations.

Reply:
Thanks for looking into this!